examples/fly_camera/main.rs

```diff
@@ -0,0 +1,164 @@
+
+        world
+            .create_entity()
+            .with(assets.cube.clone())
+            .with(assets.red.clone())
+            .with(trans)
+            .with(GlobalTransform::default())
+            .build();
+
+        let mut msg = WindowMessages::new();
+        {
+            grab_cursor(&mut msg);
+            set_mouse_cursor_none(&mut msg);
+        }
+        world.add_resource(msg);
+
+        world.add_resource(AmbientColor(Rgba::from([0.01; 3])));
+    }
+
+    fn handle_event(&mut self, _: &mut World, event: Event) -> Trans {
+        match event {
+            Event::WindowEvent { event, .. } => match event {
+                WindowEvent::KeyboardInput {
+                    input:
+                        KeyboardInput {
+                            virtual_keycode,
+                            state: ElementState::Pressed,
+                            ..
+                        },
+                    ..
+                } => match virtual_keycode {
+                    Some(VirtualKeyCode::Escape) => return Trans::Quit,
+                    _ => (),
+                },
+                _ => (),
+            },
+            _ => (),
+        }
+        Trans::None
+    }
+}
+
+struct Assets {
+    cube: MeshHandle,
+    red: Material,
+}
+
+fn load_assets(world: &World) -> Assets {
+    let mesh_storage = world.read_resource();
+    let tex_storage = world.read_resource();
+    let mat_defaults = world.read_resource::<MaterialDefaults>();
+    let loader = world.read_resource::<Loader>();
+
+    let red = loader.load_from_data([1.0, 0.0, 0.0, 1.0].into(), (), &tex_storage);
+    let red = Material {
+        albedo: red,
+        ..mat_defaults.0.clone()
+    };
+
+    let cube = loader.load("mesh/cube.obj", ObjFormat, (), (), &mesh_storage);
+
+    Assets { cube, red }
+}
+
+fn main() {
+    if let Err(error) = run() {
+        eprintln!("Could not run the example!");
+        eprintln!("{}", error);
+        ::std::process::exit(1);
+    }
+}
+
+/// Wrapper around the main, so we can return errors easily.
+fn run() -> Result<(), Error> {
+    let resources_directory = format!("{}/examples/assets", env!("CARGO_MANIFEST_DIR"));
+
+    let display_config_path = format!(
+        "{}/examples/fly_camera/resources/display_config.ron",
+        env!("CARGO_MANIFEST_DIR")
+    );
+
+    let display_config = DisplayConfig::load(display_config_path);
+
+    let key_bindings_path = format!(
+        "{}/examples/fly_camera/resources/input.ron",
+        env!("CARGO_MANIFEST_DIR")
+    );
+
+    let pipeline_builder = Pipeline::build().with_stage(
+        Stage::with_backbuffer()
+            .clear_target([0.0, 0.0, 0.0, 1.0], 1.0)
+            .with_pass(DrawShaded::<PosNormTex>::new()),
+    );
+    let mut game = Application::build(resources_directory, ExampleState)?
+        .with(
+            FlyCameraMovementSystem::new(1.0, Some("move_x"), Some("move_y"), Some("move_z")),
```

Comment:
Should provide a bundle for adding a fly camera.

Reply:
done